src/tools/kdb/command.h

```diff
@@ -0,0 +1,177 @@
+/**
+ * @file
+ *
+ * @brief Header for things used everywhere in the kdb tool
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#ifndef ELEKTRA_KDB_COMMAND_H
+#define ELEKTRA_KDB_COMMAND_H
+
+#include <colors.h>
+
+#include <kdb.h>
+#include <stdbool.h>
+#include <unistd.h>
+
+#define CLI_BASE_KEY "/sw/elektra/kdb/#0/current"
+
+#define ADD_BASIC_OPTIONS(baseSpec, baseKeyName)                                                                                           \
+	ksAppendKey (baseSpec, keyNew (baseKeyName "/debug", KEY_META, "description",                                                      \
+				       "Give debug information or ask debug questions (in interactive mode).", KEY_META, "opt", "d",       \
+				       KEY_META, "opt/long", "debug", KEY_META, "opt/arg", "none", KEY_END));                              \
+	ksAppendKey (baseSpec, keyNew (baseKeyName "/verbose", KEY_META, "description", "Explain what is happening.", KEY_META, "opt",     \
+				       "v", KEY_META, "opt/long", "verbose", KEY_META, "opt/arg", "none", KEY_END));                       \
+	ksAppendKey (baseSpec, keyNew (baseKeyName "/version", KEY_META, "description", "Print version info.", KEY_META, "opt", "V",       \
+				       KEY_META, "opt/long", "version", KEY_META, "opt/arg", "none", KEY_END));                            \
+	ksAppendKey (baseSpec, keyNew (baseKeyName "/color", KEY_META, "description", "Print never/auto(default)/always colored output.",  \
+				       KEY_META, "opt", "C", KEY_META, "opt/arg/help", "WHEN", KEY_META, "opt/long", "color", KEY_META,    \
+				       "opt/arg", "required", KEY_END));                                                                   \
+	ksAppendKey (baseSpec,                                                                                                             \
+		     keyNew (baseKeyName "/nonewline", KEY_META, "description", "Suppress the newline at the end of the output.",          \
+			     KEY_META, "opt", "n", KEY_META, "opt/long", "no-newline", KEY_META, "opt/arg", "none", KEY_END));             \
+	ksAppendKey (spec, keyNew (COMMAND_SPEC_KEY (COMMAND_NAME) "/nullterm", KEY_META, "description", "Use binary 0 termination.",      \
+				   KEY_META, "opt", "0", KEY_META, "opt/long", "null", KEY_META, "opt/arg", "none", KEY_END));
+
+#define GET_OPT_KEY(options, key) ksLookupByName (options, key, 0)
```

Comment:
Just noticed, I think this should be:

```suggestion
#define GET_OPT_KEY(options, key) ksLookupByName (options, CLI_BASE_KEY key, 0)
```

In which case use a macro makes sense, because otherwise you can't build the name like that...

It's used as e.g. `GET_OPTION_KEY (options, "debug")`, so unless I'm completely on the wrong path that would lead to `ksLookupByName (options, "debug", 0)` which always returns `NULL`, since `"debug"` is not a valid keyname.

Is there no test that verifies that CLI options (like `--debug`) actually work?

Reply:
This macro expects the whole key name since it also needs the command name, `GET_OPT_KEY (options, COMMAND_BASE_KEY (COMMAND_NAME) "/" name)` is how it is used.

Reply:
In that case `GET_BASIC_OPTIONS` is wrong, because I copied the `"debug"` example from there.

Also, please add doxygen comments to the macros that are supposed to be used by command implementations. If there was a comment stating what should be passed as `key`, it would have been quite obvious where the mistake is.